### PR TITLE
Migrate to Jakarta EE

### PR DESCRIPTION
**Pull request by mabartos**
Used commit from @adamretter : https://github.com/greenmail-mail-test/greenmail/pull/496

### CODE REVIEW

**Comment by marcelmay** [comment]
Thanks, @adamretter  and @mabartos !

JM 2.1 would be for GreenMail 2.1 . 

I will try to finalize GreenMai 2.0 soon and then merge.

**Comment by mabartos** [comment]
@marcelmay Ok, thanks.

**Comment by marcelmay** [comment]
@mabartos : I plan to release 1.6.13 this week, and 2.0.0 ~ in the next two weeks. Then I could do an 2.1-alpha-1 with your changes and JDK 11.

**Comment by mabartos** [comment]
@marcelmay Ok, thanks.

File: pom.xml [hunk 1]
```diff
@@ -424,18 +424,18 @@
...
         <groupId>jakarta.mail</groupId>
         <artifactId>jakarta.mail-api</artifactId>
-        <version>2.1.0</version>
+        <version>2.1.1</version>
       </dependency>
       <dependency>
         <groupId>org.eclipse.angus</groupId>
         <artifactId>jakarta.mail</artifactId>
-        <version>2.0.1</version>
+        <version>1.1.0</version>
```
Comment (marcelmay) [inline pom.xml:432]: Version 2.0.1

**Comment by marcelmay** [comment]
@mabartos , awfully sorry but I accidentally presumed some of your changes in 56e4554.

I tried to resolve some of the resulting conflcts in fde5251, but it would be cleaner to rebase freshly on current master (without fde5251).

The checks possibly fail currently due to  org.eclipse.angus:jakarta.mail:1.1.0  (see my comment)...

**Comment by marcelmay** [comment]
Cherry-picked : ae8c386

**Pull request closed without merging**